### PR TITLE
Update usaco-1017.mdx

### PR DESCRIPTION
**Pull request by RedBlueBird**
AP season is finally over

_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-1017.mdx [hunk 1]
```diff
@@ -15,6 +15,76 @@ author: Sofia Yang
...
+	for (int i = 0; i < c; i++){
+		int a, b, c;
+		fin >> a >> b >> c;
+		graph[a].push_back(make_pair(b,c));
+	}
+
+	//Toposort the milking sessions
+	//Start from the earliest milking session,
+	//Greedily set all its children's time to be at least itself's time + edge weight
+	//The input variable M becomes irrelevant
```
Comment (SansPapyrus683) [inline solutions/usaco-1017.mdx:61]: multiline comment

File: solutions/usaco-1017.mdx [hunk 2]
```diff
@@ -15,6 +15,76 @@ author: Sofia Yang
...
+	fin >> n >> m >> c;
+	vector<int> start(n+1);
+	for (int i = 1; i <= n; i++){
+		fin >> start[i];
+	}
+	vector<vector<pair<int, int>>> graph(n + 1);
+	for (int i = 0; i < c; i++){
+		int a, b, c;
+		fin >> a >> b >> c;
+		graph[a].push_back(make_pair(b,c));
```
Comment (SansPapyrus683) [inline solutions/usaco-1017.mdx:55]: (optional) make nodes 0-indexed

File: solutions/usaco-1017.mdx [hunk 3]
```diff
@@ -15,6 +15,76 @@ author: Sofia Yang
...
+	}
+	for (int i = n - 1; i >= 0; i--){
+		for (auto j: graph[toposort[i]]){
+			int a,b;
+			tie(a,b) = j;
+			start[a] = max(start[a], start[toposort[i]] + b);
+		}
+	}
+
+	//Output
```
Comment (SansPapyrus683) [inline solutions/usaco-1017.mdx:76]: comments need not state the obvious

File: solutions/usaco-1017.mdx [hunk 4]
```diff
@@ -15,6 +15,76 @@ author: Sofia Yang
...
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+typedef long long ll;
+using namespace std;
+
+//Construct a topologically sorted array from this function
+void dfs(vector<vector<pair<int,int>>>& graph, vector<bool>& visited, vector<int>& toposort, int node){
```
Comment (SansPapyrus683) [inline solutions/usaco-1017.mdx:26]: just call it `topo_sort` & remove the comment

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1017.mdx (hunk 2 above)
Comment (RedBlueBird) [reply solutions/usaco-1017.mdx:54]: ok, 1-indexing makes the code looks bit nicer, no need to put `-1` everywhere

Comment (SansPapyrus683) [reply solutions/usaco-1017.mdx:54]: really? it's just a `--` to all the indices?

File: solutions/usaco-1017.mdx [hunk 5]
```diff
@@ -15,6 +15,76 @@ author: Sofia Yang
...
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+typedef long long ll;
+using namespace std;
+
+void topo_sorting(vector<vector<pair<int,int>>>& graph, vector<bool>& visited, vector<int>& toposort, int node){
```
Comment (SansPapyrus683) [inline solutions/usaco-1017.mdx:25]: 80 char limit

File: solutions/usaco-1017.mdx [hunk 6]
```diff
@@ -15,6 +15,76 @@ author: Sofia Yang
...
+	visited[node] = true;
+	for (auto i: graph[node]){
+		int a,b;
+		tie(a,b) = i;
+		if (visited[a]) continue;
```
Comment (SansPapyrus683) [inline solutions/usaco-1017.mdx:30]: braces

File: solutions/usaco-1017.mdx [hunk 7]
```diff
@@ -15,6 +15,76 @@ author: Sofia Yang
...
+		dfs(graph, visited, toposort, a);
```
Comment (SansPapyrus683) [inline solutions/usaco-1017.mdx:31]: dfs?

File: solutions/usaco-1017.mdx [hunk 8]
```diff
@@ -15,6 +15,76 @@ author: Sofia Yang
...
+}
+
+
+int main() {
+	ofstream fout("timeline.out");
+	ifstream fin("timeline.in");
+
+	//Take inputs
+	//Construct an array that track all the start time
+	//Construct an adjacency list graph
```
Comment (SansPapyrus683) [inline solutions/usaco-1017.mdx:43]: multiline here

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1017.mdx (hunk 2 above)
Comment (RedBlueBird) [reply solutions/usaco-1017.mdx:58]: I mean, is it really necessary to write `--` everywhere when you don't have to

Comment (SansPapyrus683) [reply solutions/usaco-1017.mdx:58]: everywhere?

**Comment by SansPapyrus683** [comment]
i just did some lil changes myself

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/usaco-1017.mdx [hunk 9]
```diff
@@ -14,7 +14,77 @@ author: Sofia Yang
...
+```cpp
+#include <bits/stdc++.h>
+typedef long long ll;
+using namespace std;
+
+void topo_sorting(
+	vector<vector<pair<int, int>>>& graph,
+	vector<bool>& visited,
+	vector<int>& toposort, int node
+){
```
Comment (ryanchou-dev) [inline solutions/usaco-1017.mdx:28]:
```suggestion
) {
```

**Review by ryanchou-dev — APPROVED** [review]
🍵

File: solutions/usaco-1017.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/usaco-1017.mdx:55]: bump

**Review by TruaShamu — APPROVED** [review]
looks good to me :)

**Pull request merged**